### PR TITLE
Update ws-security saml.4 fat project to run with jakarta ee9 features

### PR DESCRIPTION
**Pull request by meiaus**
update com.ibm.ws.wssecurity_fat.wsscxf.saml.4 with jakarta ee9

### CODE REVIEW

**Comment by meiaus** [comment]
#run-libby-bot

**Comment by meiaus** [comment]
#build

File: dev/com.ibm.ws.wssecurity_fat.wsscxf.saml.4/fat/src/com/ibm/ws/wssecurity/fat/cxf/samltoken4/FATSuite.java [hunk 1]
```diff
@@ -37,12 +37,9 @@
...
-    //public static RepeatTests r = RepeatTests.withoutModification().andWith(FeatureReplacementAction.EE8_FEATURES().removeFeature("jsp-2.2").removeFeature("jaxws-2.2").removeFeature("servlet-3.1").removeFeature("usr:wsseccbh-1.0").addFeature("jsp-2.3").addFeature("jaxws-2.3").addFeature("servlet-4.0").addFeature("usr:wsseccbh-2.0"));
-
+	//The following run EE7 and EE8 full fat and no EE9 lite fat
     @ClassRule
-    public static RepeatTests r = RepeatTests.with(new EmptyAction().fullFATOnly()).andWith(FeatureReplacementAction.EE8_FEATURES().removeFeature("jsp-2.2").removeFeature("jaxws-2.2").removeFeature("servlet-3.1").removeFeature("usr:wsseccbh-1.0").addFeature("jsp-2.3").addFeature("jaxws-2.3").addFeature("servlet-4.0").addFeature("usr:wsseccbh-2.0"));
+    public static RepeatTests r = RepeatTests.with(new EmptyAction().fullFATOnly()).andWith(FeatureReplacementAction.EE8_FEATURES().fullFATOnly().removeFeature("jsp-2.2").removeFeature("jaxws-2.2").removeFeature("servlet-3.1").removeFeature("usr:wsseccbh-1.0").addFeature("jsp-2.3").addFeature("jaxws-2.3").addFeature("servlet-4.0").addFeature("usr:wsseccbh-2.0")).andWith(FeatureReplacementAction.EE9_FEATURES());
```
Comment (jimmy1wu) [inline dev/com.ibm.ws.wssecurity_fat.wsscxf.saml.4/fat/src/com/ibm/ws/wssecurity/fat/cxf/samltoken4/FATSuite.java:42]: should we be explicit about the features removed/added for ee9?

from cxf saml.1 and saml.2:
```
    public static RepeatTests r = RepeatTests.with(new EmptyAction().fullFATOnly()).andWith(FeatureReplacementAction.EE8_FEATURES().fullFATOnly().removeFeature("jsp-2.2").removeFeature("jaxws-2.2").removeFeature("servlet-3.1").removeFeature("usr:wsseccbh-1.0").addFeature("jsp-2.3").addFeature("jaxws-2.3").addFeature("servlet-4.0").addFeature("usr:wsseccbh-2.0")).andWith(FeatureReplacementAction.EE9_FEATURES().removeFeature("jsp-2.3").removeFeature("jaxws-2.3").removeFeature("servlet-4.0").removeFeature("usr:wsseccbh-1.0").addFeature("pages-3.0").addFeature("xmlWS-3.0").addFeature("servlet-5.0").addFeature("usr:wsseccbh-2.0"));
```

Comment (meiaus) [reply dev/com.ibm.ws.wssecurity_fat.wsscxf.saml.4/fat/src/com/ibm/ws/wssecurity/fat/cxf/samltoken4/FATSuite.java:42]: As mentioned in the slack, saml.3, saml.4, saml.5 FAT do not contain lite buckets except the dummy one, and these FAT will run EE7 and EE8 Full only; therefore no ee9 feature replacement is used here.

Basically, the invocation of `andWith(FeatureReplacementAction.EE9_FEATURES()` will still attempt to run EE9 test as logged in output.txt:
```
[07/27/2021 19:36:01:127 PDT] 001 CompositeRepeatTestActionState evaluate                       I Running tests with action: JakartaEE9 FAT repeat action
```
but the test classes in the test suite are already set to skip ee9 `@SkipForRepeat({ EE9_FEATURES })`, so no feature replacement to be called out explicitly in the class rule.  output.txt shows the skipping: (taking saml.5 FAT as an example)
```
[07/27/2021 19:36:50:173 PDT] 001 RepeatTestFilter               shouldRun                      I Skipping test class com.ibm.ws.wssecurity.fat.cxf.samltoken5.TwoServerTests.CxfSAMLSymSignEnc2ServerTests on action EE9_FEATURES
[07/27/2021 19:36:50:173 PDT] 001 RepeatTestFilter               shouldRun                      I Skipping test class com.ibm.ws.wssecurity.fat.cxf.samltoken5.TwoServerTests.CxfSAMLAsymSignEnc2ServerTests on action EE9_FEATURES
[07/27/2021 19:36:50:173 PDT] 001 RepeatTestFilter               shouldRun                      I Skipping test class com.ibm.ws.wssecurity.fat.cxf.samltoken5.OneServerTests.CxfSAMLSymSignEnc1ServerTests on action EE9_FEATURES
[07/27/2021 19:36:50:173 PDT] 001 RepeatTestFilter               shouldRun                      I Skipping test class com.ibm.ws.wssecurity.fat.cxf.samltoken5.OneServerTests.CxfSAMLAsymSignEnc1ServerTests on action EE9_FEATURES

```

**Review by jimmy1wu — APPROVED** [review]

**Comment by arunavemulapalli** [comment]
for #12449

**Pull request merged**